astroquery/heasarc/core.py

```diff
@@ -1,92 +1,247 @@
-# Licensed under a 3-clause BSD style license - see LICENSE.rst
 
-from typing import Union
+import os
+
+import shutil
+import requests
+import tarfile
 import warnings
-from io import StringIO, BytesIO
+import numpy as np
 from astropy.table import Table
-from astropy.io import fits
 from astropy import coordinates
 from astropy import units as u
-from ..query import BaseQuery
-from ..utils import commons
-from ..utils import async_to_sync
+from astropy.utils.decorators import deprecated, deprecated_renamed_argument
+
+import pyvo
+
+from astroquery import log
+from ..query import BaseQuery, BaseVOQuery
+from ..utils import commons, async_to_sync, parse_coordinates
 from ..exceptions import InvalidQueryError, NoResultsWarning
 from . import conf
 
-__all__ = ['Heasarc', 'HeasarcClass']
 
+@async_to_sync
+class HeasarcClass(BaseVOQuery, BaseQuery):
+    """Class for accessing HEASARC data using XAMIN.
 
-def Table_read(*args, **kwargs):
-    if commons.ASTROPY_LT_5_1:
-        return Table.read(*args, **kwargs)
-    else:
-        return Table.read(*args, **kwargs, unit_parse_strict='silent')
+    Example Usage:
+    ...
 
+    """
 
-@async_to_sync
-class HeasarcClass(BaseQuery):
+    # we can move url to Config later
+    VO_URL = conf.VO_URL
+    TAR_URL = conf.TAR_URL
+    S3_BUCKET = conf.S3_BUCKET
```

Comment:
Do we need these here, or the properties could pick them up? (like we do in `alma` or the other modules that use pyvo based tap?)

Reply:
Are you suggesting that these should not be configurable but rather as fixed variables?